### PR TITLE
Feature support route disabling

### LINKED ISSUE #106: Support route disabling
This is a feature request. Instead of deleting a route if I don't want it right now, can we disable it now? And we can enable it any time in the future to save some time in case need it again?

### PR DESCRIPTION
**Pull request by gabrielperes97**
**Description**
Implements a feature do disable/enable a route. When
Issue #106 

**How Has This Been Tested?**
I use the basic-data environment, start the environment and made a call to "/anwser". Then I disable the route, restart the server and made another call to "/answer", now the route is not caught. I enable the route and restart the server again and made the call, the route is caught.

**Screenshots**
Screenshot_20191026_210117
Screenshot_20191026_210140

**Closing issues**
Closes #106

### CODE REVIEW

**Comment by JoshMLeslie** [comment]
I agree with the @255kb 's comments about the UI - what about having a hover effect on the action tag that shows a power button? Then when it's off it could go to a 50% opacity.

[icon link](https://material.io/resources/icons/?search=power&icon=power_settings_new&style=baseline)

**on**
image

**off**
image

**Comment by 255kb** [comment]
@JoshMLeslie I like the design, but I still think that it's not something so widely used that you need the button to be always there. For me a right click menu item should be enough. 
I really want to avoid UI overloading, also considering that a lot of more important options/elements will be added in the UI (disable templating, add websocket route, routes grouping, etc).
But I am still open to discuss this 😃

**Comment by gabrielperes97** [comment]
I agree that the disable button is overloading the UI, I will move this to context menu. But I think the colored corner is a bit strange when it is over the left shadow. What if we use a red left shadow to indicate the disabled state?

Screenshot_20191030_201442

**Comment by 255kb** [comment]
Sure forget about the red corner 🙂
I like the red shadow, why not put it on the right, so we can still see the blue one when the route is selected?

File: src/app/services/server.service.ts [hunk 1]
```diff
@@ -152,7 +152,7 @@ export class ServerService {
       const duplicatedRoutes = this.store.get('duplicatedRoutes')[environment.uuid];
 
       // only launch non duplicated routes
-      if (!duplicatedRoutes || !duplicatedRoutes.has(declaredRoute.uuid)) {
+      if ((!duplicatedRoutes || !duplicatedRoutes.has(declaredRoute.uuid)) && declaredRoute.enabled !== false) {
```
[the remark this answers is not in the document]
Comment (gabrielperes97) [reply src/app/services/server.service.ts]: Done

**Comment by gabrielperes97** [comment]
Done

**Comment by 255kb** [comment]
Thank you for the modification @gabrielperes97 :)

**Pull request merged**